### PR TITLE
config: allow reloading of telemetry settings

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
This updates the way we handle configuration to support reloading without restarting for the following options:

- `debug`
- `log_level`
- `proxy_log_level`
- `metrics_address`
- `tracing_provider`
- `tracing_sample_rate`
- `tracing_jaeger_collector_endpoint`
- `tracing_jaeger_agent_endpoint`
- `tracing_zipkin_endpoint`

I also had to move the tracing options from config to trace to avoid an import cycle.

## Related issues
- #1254 

**Checklist**:
- [x] add related issues
- [ ] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

**Comment by travisgroth** [comment]
FYI - for the tracing options to become fully dynamic, we'll need to move the envoy tracing config from the [bootstrap](https://github.com/pomerium/pomerium/blob/master/internal/envoy/envoy.go#L264-L310) file into the control plane.  This wasn't possible until envoy 1.15.

File: internal/envoy/envoy.go [hunk 1]
```diff
@@ -64,12 +64,6 @@ func NewServer(opts *config.Options, grpcPort, httpPort string) (*Server, error)
 		opts:     opts,
 	}
```
Comment (desimone) [inline internal/envoy/envoy.go:66]: Should we update the docs https://www.pomerium.io/reference/#proxy-log-level ?

File: internal/envoy/envoy.go [hunk 2]
```diff
@@ -88,7 +82,7 @@ func (srv *Server) Run(ctx context.Context) error {
 
 	srv.cmd = exec.CommandContext(ctx, envoyPath,
 		"-c", configFileName,
-		"--log-level", srv.logLevel,
+		"--log-level", "trace",
```
Comment (desimone) [inline internal/envoy/envoy.go:85]: Will this be updated by config/log.go or is that just for the main log level? Is the envoy log level now hardcoded to trace?

**Comment by desimone** [comment]
\cc @travisgroth because this is his area :)

File: internal/envoy/envoy.go (hunk 2 above)
Comment (calebdoxsey) [reply internal/envoy/envoy.go:85]: this is so all the logs are available in the handler. Whether they are displayed or not is a result of the zerolog log level.

**Comment by calebdoxsey** [comment]
@travisgroth apparently its still not supported:

> Please be aware that envoy.tracers.opencensus provider can only be configured once in Envoy lifetime. Any attempts to reconfigure it or to use different configurations for different HCM filters will be rejected. Such a constraint is inherent to OpenCensus itself. It cannot be overcome without changes on OpenCensus side.

File: internal/envoy/envoy.go (hunk 1 above)
Comment (calebdoxsey) [reply internal/envoy/envoy.go:66]: It didn't really change. I updated the docs to make it clear the setting only applies to access logs.

File: internal/envoy/envoy.go (hunk 2 above)
Comment (desimone) [reply internal/envoy/envoy.go:85]: @calebdoxsey perfect

**Comment by travisgroth** [comment]
> @travisgroth apparently its still not supported:
> 
> > Please be aware that envoy.tracers.opencensus provider can only be configured once in Envoy lifetime. Any attempts to reconfigure it or to use different configurations for different HCM filters will be rejected. Such a constraint is inherent to OpenCensus itself. It cannot be overcome without changes on OpenCensus side.

🙄  that's awesome.  okay.

File: internal/envoy/envoy.go (hunk 2 above)
Comment (travisgroth) [reply internal/envoy/envoy.go:85]: I recall we had some performance problems when we initial switched over to envoy.  https://github.com/pomerium/pomerium/pull/779 was pulling in a default that, I believe, set us to trace and slowed down performance significantly.  I might be misremembering but we should check this.

**Pull request merged**